source/gameengine/Ketsji/KX_SG_NodeRelationships.cpp

```diff
@@ -62,15 +65,20 @@ bool KX_NormalParentRelation::UpdateChildCoordinates(SG_Node *child,
     return true;  // false;
   }
   else {
-    // the childs world locations which we will update.
-    const MT_Vector3 &p_world_scale = parent->GetWorldScaling();
-    const MT_Vector3 &p_world_pos = parent->GetWorldPosition();
-    const MT_Matrix3x3 &p_world_rotation = parent->GetWorldOrientation();
-
-    child->SetWorldScale(p_world_scale * child->GetLocalScale());
-    child->SetWorldOrientation(p_world_rotation * child->GetLocalOrientation());
-    child->SetWorldPosition(p_world_pos +
-                            p_world_scale * (p_world_rotation * child->GetLocalPosition()));
+    const MT_Transform trans(parent->GetWorldTransform() * child->GetLocalTransform());
+    MT_Matrix4x4 tmat(trans.toMatrix());
+    float sx = MT_Vector3(tmat.getElement(0, 0), tmat.getElement(1, 0), tmat.getElement(2, 0)).length();
+    float sy = MT_Vector3(tmat.getElement(0, 1), tmat.getElement(1, 1), tmat.getElement(2, 1)).length();
+    float sz = MT_Vector3(tmat.getElement(0, 2), tmat.getElement(1, 2), tmat.getElement(2, 2)).length();
```

Comment:
Cannot use tmat.GetWorldScaling() instead?

Reply:
tmat has no function GetWorldScaling.
I did as in upbge 0.2.5: get a transform matrix and extract position, rotation, and scale. In 0.2.5, there was a method to extract directly the scale but in 0.3 the maths lib is Moto, and i had to do it following the article i mentionned in the pull request.